### PR TITLE
CONSOLE-3777: Update NodeJS v14 to v18

### PR DESCRIPTION
**Pull request by cyril-ui-developer**

### CODE REVIEW

**Review by jhadvig — COMMENTED** [review]
Looks good 👍 
@cyril-ui-developer please update the README accordingly

**Comment by cyril-ui-developer** [comment]
@jhadvig I have another [PR](https://github.com/openshift/release/pull/44053) in the release repo to update the mapping for `quay.io/coreos/tectonic-console-builder` image from v25 to v26. PTAL

**Comment by cyril-ui-developer** [comment]
/

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Comment by cyril-ui-developer** [comment]
/

File: README.md [hunk 1]
```diff
@@ -108,6 +108,25 @@ In order to enable the monitoring UI and see the "Observe" navigation item while
...
+   - Dockerfile.dev
+   - Dockerfile.plugins.demo
+   - Dockerfile.plugins.demo2
+   For example, `tectonic-console-builder:27`
+2. Update the dependencies in Dockerfile.builder file i.e. v18.0.0
+3. Run `./push-builder.sh` script build and push the  updated builder image to quay.io.
+   Note: You can test the image using `./builder-run.sh ./build-backend.sh`.
+         Also, to update the image on quay.io, you need edit permission to the quay.io/coreos/tectonic-console-builder repo. 
+4. Lastly, update the mapping of `tectonic-console-builder` image tag in [release-repo](https:// github.com/openshift/release/blob/master/core-services/image-mirroring/supplemental-ci-images/mapping_supplemental_ci_images_ci).
+   Note: There could be scenario were you would have to add the new image reference in the "mapping_supplemental_ci_images_ci" file, i.e. to avoid CI downtime for upcoming release cycle.
```
[the remark this answers is not in the document]
Comment (cyril-ui-developer) [reply README.md:127]: @jhadvig Thanks for feedback. But I have two questions. 
1. Do we have to remove the old version mapping  at some point in time in the future. Asking bcos there is no `v24` in the [release repo mapping file](https://github.com/cyril-ui-developer/release/blob/2a835fcb0881cae3ca81f4fff6e247d63d38a54f/core-services/image-mirroring/supplemental-ci-images/mapping_supplemental_ci_images_ci#L111).
2. The [supplimental_ci_mapping file](https://github.com/openshift/release/blob/master/clusters/app.ci/supplemental-ci-images/origin-stable/release.yaml) has  reference for `"quay.io/coreos/tectonic-console-builder:v24"` but don't have for `v25`. Do we need to update this file as part of the changes in this PR?

Could you please clarify?

**Comment by cyril-ui-developer** [comment]
N.B. I am seeing the error below in the browser console:
`WebSocketClient.js:13 WebSocket connection to 'ws://localhost:8080/ws' failed: 
WebSocketClient @ WebSocketClient.js:13
initSocket @ socket.js:27
(anonymous) @ socket.js:51
index.js:493 [webpack-dev-server] Event {isTrusted: true, type: 'error', target: WebSocket, currentTarget: WebSocket, eventPhase: 2, …}` 

Looking into it....

**Comment by jhadvig** [comment]
Hmm not sure why the tectonic builder contains old version but after pulling `quay.io/coreos/tectonic-console-builder:v26` locally I see v18.18.1
/retest

**Comment by jhadvig** [comment]
OK I see the issue. It looks like `registry.ci.openshift.org/ocp/builder:rhel-8-base-nodejs-openshift-4.15` contains node v16. We need bump that version.

**Comment by cyril-ui-developer** [comment]
> registry.ci.openshift.org/ocp/builder:rhel-8-base-nodejs-openshift-4.15

@jhadvig I have been observing the failure but not sure what is going on. If the issue is `registry.ci.openshift.org/ocp/builder:rhel-8-base-nodejs-openshift-4.15` contains node v16. was it update to node.jsv16 recently, and how come the current builder image -`tectonic-console-builder:v25` based on node.js-v14 runs without failure? 

I am wondering why [release.yaml](https://github.com/openshift/release/blob/master/clusters/app.ci/supplemental-ci-images/origin-stable/release.yaml#L100) is missing `quay.io/coreos/tectonic-console-builder:v25"` which could have been the most recent update based on builder image in quay.io prior to my PR change.. However seems it doesn't impact console UI tests since we are not experiencing any docker build failure.

I am thinking we need to update  [release.yaml](https://github.com/openshift/release/blob/master/clusters/app.ci/supplemental-ci-images/origin-stable/release.yaml#L100) with the latest builder image tag - v26 but not sure if that would fix the issue.

File: README.md [hunk 2]
```diff
@@ -108,6 +108,25 @@ In order to enable the monitoring UI and see the "Observe" navigation item while
...
+#### Updating `tectonic-console-builder` image
+Updating `tectonic-console-builder` image is needed whenever there is a change in the build-time dependencies and/or go versions.
+
+In order to update the `tectonic-console-builder` to a new version i.e. v27, follow these steps:
+
+1. Update the `tectonic-console-builder` image tag in files listed below:
+   - .ci-operator.yaml
+   - Dockerfile.dev
+   - Dockerfile.plugins.demo
+   - Dockerfile.plugins.demo2
```
Comment (jhadvig) [inline README.md:120]: nit: this dockerfile is not using `tectonic-console-builder` image

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (petr-muller) [reply README.md:126]: sry I'm out of CI matters for too long now to correctly advise on this :(

**Comment by rhamilto** [comment]
@cyril-ui-developer, [Protractor and webdriver have been removed](https://github.com/openshift/console/pull/13284). :)

**Comment by cyril-ui-developer** [comment]
> @cyril-ui-developer, [Protractor and webdriver have been removed](https://github.com/openshift/console/pull/13284). :)

Awesome! Thank you for the heads up.

**Comment by rhamilto** [comment]
@cyril-ui-developer, looks like there is an issue manifesting as an issue with the Cypress config resulting from these changes.  https://search.ci.openshift.org/?search=Your+configFile+threw+an+error+from%3A+cypress.config.js&maxAge=168h&context=1&type=all&name=pull-ci-openshift-console-master-e2e-gcp-console&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

**Comment by cyril-ui-developer** [comment]
> @cyril-ui-developer, looks like there is an issue manifesting as an issue with the Cypress config resulting from these changes. https://search.ci.openshift.org/?search=Your+configFile+threw+an+error+from%3A+cypress.config.js&maxAge=168h&context=1&type=all&name=pull-ci-openshift-console-master-e2e-gcp-console&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

Correct, I have seen it. It is related to the `hashFunction` issue which I fixed in webpack.config. Added the same fix to cypress-config. Retesting to see if that fix it. It looks good in local. Thanks.

**Comment by jcaianirh** [comment]
Work is tech debt and has no customer-facing impact.

/label docs-approved
/label px-approved
/label qe-approved

**Pull request merged**